adapters/clydo/clydo.go

```diff
@@ -0,0 +1,294 @@
+package clydo
+
+import (
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse endpoint url: %v", err)
+	}
+	bidder := &adapter{
+		endpoint: template,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
+
+	for _, imp := range request.Imp {
+		reqData, err := a.prepareRequest(request, imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		requests = append(requests, reqData)
+	}
+	// fmt.Println(errors)
+	return requests, errors
+}
+
+func (a *adapter) MakeBids(
+	request *openrtb2.BidRequest,
+	requestData *adapters.RequestData,
+	responseData *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
+	if errResp := checkResponseStatus(responseData); errResp != nil {
+		return nil, errResp
+	}
+	response, err := prepareBidResponse(responseData.Body)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	if response.Cur != "" {
+		bidResponse.Currency = response.Cur
+	} else {
+		bidResponse.Currency = "USD"
```

Comment:
The `Currency` field is already set the default value of `"USD"` inside of `NewBidderResponseWithBidsCapacity()`. The `else` clause could be removed:
```
61     bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
62     if response.Cur != "" {
63         bidResponse.Currency = response.Cur
64 -   } else {
65 -       bidResponse.Currency = "USD"
66     }
adapters/clydo/clydo.go
```